wait/http_test.go

```diff
@@ -25,7 +36,88 @@ func ExampleHTTPStrategy() {
 		panic(err)
 	}
 
-	defer gogs.Terminate(ctx)
-
+	defer gogs.Terminate(ctx) // nolint: errcheck
 	// Here you have a running container
+
+}
+
+func TestHTTPStrategyWaitUntilReady(t *testing.T) {
+	workdir, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	capath := workdir + "/testdata/root.pem"
+	cafile, err := ioutil.ReadFile(capath)
+	if err != nil {
+		t.Errorf("can't load ca file: %v", err)
+		return
+	}
+
+	certpool := x509.NewCertPool()
+	if !certpool.AppendCertsFromPEM(cafile) {
+		t.Errorf("the ca file isn't valid")
+		return
+	}
+
+	tlsconfig := &tls.Config{RootCAs: certpool, ServerName: "testcontainer.go.test"}
+	dockerReq := testcontainers.ContainerRequest{
+		FromDockerfile: testcontainers.FromDockerfile{
+			Context: workdir + "/testdata",
+		},
+		ExposedPorts: []string{"6443/tcp"},
+		WaitingFor: wait.NewHTTPStrategy("/ping").WithTLS(true, tlsconfig).
+			WithStartupTimeout(time.Second * 10).WithPort("6443/tcp").
+			WithResponseMatcher(func(body io.Reader) bool {
+				data, _ := ioutil.ReadAll(body)
+				return bytes.Equal(data, []byte("pong"))
+			}).
+			WithMethod(http.MethodPost).WithBody(bytes.NewReader([]byte("ping"))),
+	}
+
+	container, err := testcontainers.GenericContainer(context.Background(),
+		testcontainers.GenericContainerRequest{ContainerRequest: dockerReq, Started: true})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer container.Terminate(context.Background()) // nolint: errcheck
+
+	host, err := container.Host(context.Background())
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	port, err := container.MappedPort(context.Background(), "6443/tcp")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	client := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsconfig,
+			Proxy:           http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		},
+	}
+	resp, err := client.Get(fmt.Sprintf("https://%s:%s", host, port.Port()))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
```

Comment:
What do you think about adding an assertion for the `pong` response? We'd like to check that the response matcher works as intended.

Reply:
this variable indicates a state code, and I consider the value of `pong` to be a success state.

in practice, the server can return a response body with a state code representing the current server state,so we can know if the server is ready based on what is specifically returned.

Reply:
:) I meant about adding another assertion for the expected response (pong), which is part of the waitFor condition predefined when setting up the test in the response matcher. Of course this is not a requirement by the way, so I'm not opposing to it.

In the same hand, the PR as it's now could be merged, so I want to thank you for your contribution